src/tactic/core.lean

```diff
@@ -64,14 +66,32 @@ end expr
 namespace interaction_monad
 open result
 
+variables {σ : Type} {α : Type u}
+
+/-- `get_state` returns the underlying state inside an interaction monad, from within that monad. -/
+meta def get_state : interaction_monad σ σ :=
+λ state, success state state
+
+/-- `set_state` sets the underlying state inside an interaction monad, from within that monad. -/
+meta def set_state (state : σ) : interaction_monad σ unit :=
+λ _, success () state
+
+/-- `under_state` applies the given state to `tac`, and returns the result. If `tac` fails, then
+`under_state` does too. -/
+meta def under_state (state : σ) (tac : interaction_monad σ α) : interaction_monad σ α :=
+λ s, match tac state with
+     | success val _       := success val s
+     | exception fn pos ts := exception fn pos ts
+     end
+
 /-- `get_result tac` returns the result state of applying `tac` to the current state.
-Note that it does not update the current state. -/
-meta def get_result {σ α} (tac : interaction_monad σ α) :
-  interaction_monad σ (interaction_monad.result σ α) | s :=
-match tac s with
-| r@(success _ s') := success r s'
-| r@(exception _ _ s') := success r s'
-end
+Note that it updates the current state. -/
```

Comment:
I agree that `get_result` wasn't following its spec. But why not fix the implementation, rather than the spec?

Is `get_result` actually being used anywhere? If this bug wasn't causing a problem, maybe that's a sign we should just delete `get_result`.

Reply:
The reason not to do this is that the only use for `get_result` in the library is the following function `of_tactic'`, which fixes the `lean.parser` monad's handling of failures in coerced tactics.

The function as previously documented is precisely `lock_tactic_state` (and maybe this has another name somewhere else too?).

Reply:
Got it.

Reply:
It seems that `of_tactic` in core has been fixed in the meantime!

Reply:
And as such `get_result` isn't being used at all.